estela_cli/context.py

```diff
@@ -30,12 +36,73 @@ def test_host(host):
     assert "projects" in response.json()
 
 
+def prompt_context(name, username=None, password=None, host=None):
+    try:
+        if host is None:
+            host = click.prompt("Host")
+        if username is None:
+            username = click.prompt("Username")
+        if password is None:
+            password = getpass()
+
+        estela_config_path = get_estela_config_path()
+        estela_config = get_estela_config()
+
+        if estela_config is None:
+            estela_config = {}
+
+        estela_config[name] = {
+            "host": host,
+            "username": username,
+            "password": password,
+        }
+        with open(estela_config_path, "w") as estela_config_yaml:
+            yaml.dump(estela_config, estela_config_yaml)
+            click.echo(
+                "Successful login. Context {} stored in ~/{}.".format(
+                    name, ESTELA_CONFIG_NAME
+                )
+            )
+        estela_client = EstelaClient(host, username, password)
+
+    except:
+        raise Exception("Unable to login with provided credentials.")
+
+    return estela_client
+
+
 @click.command(name="context", short_help=SHORT_HELP)
-def estela_command():
+@click.argument("name", required=False)
+def estela_command(name):
     host = get_host_from_env()
     username = get_username_from_env()
     password = get_password_from_env()
 
+    if name:
+        click.echo(f"Checking your current context for {name}...")
+        estela_config = get_estela_config()
+
+        if estela_config and estela_config.get(name, None):
+            click.echo("Context {} found.".format(name))
+            try:
+                test_host(estela_config[name]["host"])
+                click.echo(OK_HOST.format(estela_config[name]["host"]))
+                estela_client = EstelaClient(
+                    estela_config[name]["host"],
+                    estela_config[name]["username"],
+                    estela_config[name]["password"],
+                )
+            except:
+                click.echo(BAD_HOST)
+                return
+        else:
+            click.echo(
+                "Context {} not found.\nInitializing context...".format(name)
+            )
+            estela_client = prompt_context(name, username, password, host)
```

Comment:
If I use incorrect credentials, I get this output:

```
Checking your current context for bad-local...
Context bad-prod not found.
Initializing context...
Host: https://127.0.0.1
Username: mgonnav
Password: 
Successful login. Context bad-prod stored in ~/.estela-config.yaml.
Traceback (most recent call last):
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/estela-0.2.7-py3.9.egg/estela_cli/context.py", line 71, in prompt_context
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/estela-0.2.7-py3.9.egg/estela_cli/estela_client.py", line 23, in __init__
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/estela-0.2.7-py3.9.egg/estela_cli/estela_client.py", line 113, in check_status
Exception: ['Unable to log in with provided credentials.']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mgonnav/.pyenv/versions/estela-cli/bin/estela", line 33, in <module>
    sys.exit(load_entry_point('estela==0.2.7', 'console_scripts', 'estela')())
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/estela-0.2.7-py3.9.egg/estela_cli/context.py", line 105, in estela_command
  File "/home/mgonnav/.pyenv/versions/estela-cli/lib/python3.9/site-packages/estela-0.2.7-py3.9.egg/estela_cli/context.py", line 74, in prompt_context
Exception: Unable to login with provided credentials.
```
You can see "Successful login" and errors because I provided incorrect credentials. If the credentials are incorrect, the context should simply not be saved.